root/tree/dataframe/df103_NanoAODHiggsAnalysis.C

```diff
@@ -0,0 +1,369 @@
+         if (dr < 0.02) {
+            return false;
+         }
+      }
+      return true;
+   };
+   auto df_z_dr = df_z_idx.Filter(filter_z_dr, {"Z_idx", "Electron_eta", "Electron_phi"},
+                                  "Delta R separation of Electrons building Z system");
+
+   // Compute masses of Z systems
+   auto df_z_mass = df_z_dr.Define("Z_mass", compute_z_masses_4l,
+                                   {"Z_idx", "Electron_pt", "Electron_eta", "Electron_phi", "Electron_mass"});
+
+   // Cut on mass of Z candidates
+   auto df_z_cut = filter_z_candidates(df_z_mass);
+
+   // Reconstruct H mass
+   auto df_h_mass = df_z_cut.Define("H_mass", compute_higgs_mass_4l,
+                                    {"Z_idx", "Electron_pt", "Electron_eta", "Electron_phi", "Electron_mass"});
+
+   return df_h_mass;
+}
+
+// Plot invariant mass for signal and background processes from simulated events
+// overlay the measured data.
+template <typename T>
+void plot(T sig, T bkg, T data, const std::string &x_label, const std::string &filename)
+{
+   // Canvas and general style options
+   gStyle->SetOptStat(0);
+   gStyle->SetTextFont(42);
+   auto c = new TCanvas("c", "", 800, 700);
+   c->SetLeftMargin(0.15);
+
+   // Get signal and background histograms and stack them to show Higgs signal
+   // on top of the background process
+   auto h_sig = *sig;
+   auto h_bkg = *bkg;
+   auto h_cmb = *(TH1D*)(sig->Clone());
+   h_cmb.Add(&h_bkg);
+   h_cmb.SetTitle("");
+   h_cmb.GetXaxis()->SetTitle(x_label.c_str());
+   h_cmb.GetXaxis()->SetTitleSize(0.04);
+   h_cmb.GetYaxis()->SetTitle("N_{Events}");
+   h_cmb.GetYaxis()->SetTitleSize(0.04);
+   h_cmb.SetLineColor(kRed);
+   h_cmb.SetLineWidth(2);
+   h_cmb.SetMaximum(18);
+
+   h_bkg.SetLineWidth(2);
+   h_bkg.SetFillStyle(1001);
+   h_bkg.SetLineColor(kBlack);
+   h_bkg.SetFillColor(kAzure - 9);
+
+   // Get histogram of data points
+   auto h_data = *data;
+   h_data.SetLineWidth(1);
+   h_data.SetMarkerStyle(20);
+   h_data.SetMarkerSize(1.0);
+   h_data.SetMarkerColor(kBlack);
+   h_data.SetLineColor(kBlack);
+
+   // Draw histograms
+   h_cmb.DrawClone("HIST");
+   h_bkg.DrawClone("HIST SAME");
+   h_data.DrawClone("PE1 SAME");
+
+   // Add legend
+   TLegend legend(0.62, 0.70, 0.82, 0.88);
+   legend.SetFillColor(0);
+   legend.SetBorderSize(0);
+   legend.SetTextSize(0.03);
+   legend.AddEntry(&h_data, "Data", "PE1");
+   legend.AddEntry(&h_bkg, "ZZ", "f");
+   legend.AddEntry(&h_cmb, "m_{H} = 125 GeV", "f");
+   legend.Draw();
+
+   // Add header
+   TLatex cms_label;
+   cms_label.SetTextSize(0.04);
+   cms_label.DrawLatexNDC(0.16, 0.92, "#bf{CMS Open Data}");
+   TLatex header;
+   header.SetTextSize(0.03);
+   header.DrawLatexNDC(0.63, 0.92, "#sqrt{s} = 8 TeV, L_{int} = 11.6 fb^{-1}");
+
+   // Save plot
+   c->SaveAs(filename.c_str());
+}
+
+void payload(unsigned int nthreads)
+{
+#ifdef R__USE_IMT
+   // Enable multi-threading
+   if (nthreads) {
```

Comment:
C++ has `if(nthreads)`, python has `if nthreads > 1`. We should probably pick one for consistency

Reply:
I think the `nthreas > 1` is what we want, I'll fix it.